src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_BinaryOperator.cs

```diff
@@ -1974,8 +1984,8 @@ private BoundExpression MakeNullCheck(SyntaxNode syntax, BoundExpression rewritt
             int rightMask)
         {
             SyntaxNode rightSyntax = loweredRight.Syntax;
-            ConstantValue rightConstantValue = loweredRight.ConstantValue;
-            TypeSymbol rightType = loweredRight.Type;
+            ConstantValue? rightConstantValue = loweredRight.ConstantValue;
+            TypeSymbol rightType = loweredRight.Type!;
```

Comment:
How do we know?

Reply:
because it is a built-in shift operator.

---
In reply to: [386602527](https://github.com/dotnet/roslyn/pull/42068#discussion_r386602527) [](ancestors = 386602527)